src/Servers/Kestrel/Core/src/Internal/Http3/Http3Stream.cs

```diff
@@ -576,7 +576,130 @@ private bool TryClose()
         return false;
     }
 
+    private async ValueTask<long> TryReadStreamHeaderAsync(Http3StreamType? advanceOn)
+    {
+        var advance = false;
+        SequencePosition consumed = default;
+        SequencePosition start = default;
+        try
+        {
+            while (_isClosed == 0)
```

Comment:
Here `_isClosed` isn't declared as `volatile`, so it should be `Volatile.Read`s and writes (L96)?

Reply:
Good catch, in `Http3Stream` it's not defined as [volatile](https://github.com/dotnet/aspnetcore/blob/28739c8e10f562c0912b9a3961592c3f1532522d/src/Servers/Kestrel/Core/src/Internal/Http3/Http3Stream.cs#L52), and I believe we should treat them the same way and make it volatile as well in `Http3Stream`.